crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -160,23 +160,29 @@ impl Instruction {
 
     /// Try to simplify this instruction. If the instruction can be simplified to a known value,
     /// that value is returned. Otherwise None is returned.
-    pub(crate) fn simplify(&self, dfg: &mut DataFlowGraph) -> Option<ValueId> {
+    pub(crate) fn simplify(&self, dfg: &mut DataFlowGraph) -> SimplifyResult {
```

Comment:
This was changed to use a new enum so that we can differentiate between the "Remove" case and "replace with a new value." Previously to remove a known constrain instruction, we'd say to replace with constant zero, but this caused asserts in the unrolling pass to fail since the result count was no longer equal to the old result count (0 != 1).